components/utilities/combination.js

```diff
@@ -0,0 +1,176 @@
+// Function validateCombo that takes in an array of cards which is the current players choice of cards, and matches it to the current cycles combination. 
+export const dictionaryCombinations = {
+  'single': {
+    // Can't be more than 1 card.
+    isValid: (cards) => cards.length === 1,
+  },
+  'pair': {
+    // Comprised of two cards, cards have the same number.
+    isValid: (cards) => {
+      if (cards.length !== 2) {
+        return false;
+      }
+    
+      return cards[0].number === cards[1].number;
+    },
+  },
+  'triplet': {
+    // Comprised of three cards, cards have the same number.
+    isValid: (cards) => {
+      if (cards.length !== 3) {
+        return false;
+      }
+    
+      return cards[0].number === cards[1].number && cards[0].number === cards[2].number;
+    }
+  },
+  'quartet': {
+    // Comprised of four cards, cards have the same number.
+    isValid: (cards) => {
+      if (cards.length !== 4) {
+        return false;
+      }
+    
+      return cards[0].number === cards[1].number && cards[0].number === cards[2].number && cards[0].number === cards[3].number;
```

Comment:
this will be the same for pair, triplet and quartet, but you can optimize it by just grabbing the first card number, then making sure each card after is the same

```javascript
const [{number: cardNumber}] = cards; // gets the first card's number
cards.every(({number}) => number === cardNumber);
```

or something to this effect. if you really wanna make it concise, something like this would work:

```javascript
isValid: (cards) => {
  const [{number: cardNumber}] = cards; // gets the first card's number
  return cards.length === 4 && cards.every(({number}) => number === cardNumber);
}
```

^^ this checks that both the length is 4 AND each card is the same in nice lil concise way.

Reply:
This is beautiful, thanks.